libtenzir/builtins/formats/syslog.cpp

```diff
@@ -0,0 +1,108 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/format/reader_factory.hpp>
+#include <tenzir/format/syslog.hpp>
+#include <tenzir/plugin.hpp>
+#include <tenzir/table_slice_builder.hpp>
+#include <tenzir/to_lines.hpp>
+
+namespace tenzir::plugins::syslog {
+
+namespace {
+
+auto impl(generator<std::optional<std::string_view>> lines,
+          operator_control_plane& ctrl) -> generator<table_slice> {
+  for (auto&& line : lines) {
+    if (not line) {
+      co_yield {};
+      continue;
+    }
+    if (line->empty()) {
+      co_yield {};
+      continue;
+    }
+    const auto* f = line->begin();
+    const auto* const l = line->end();
+    std::optional<table_slice_builder> builder{};
```

Comment:
You create (and flush) a new builder for every iteration here. This means that the outgoing batches only have 1 event each. Can you rewrite this so that the builder is only flushed when the event type changes?

Reply:
Sure.